vaccine/vaccine_reg_ussd.py

```diff
@@ -0,0 +1,363 @@
+from datetime import date
+from enum import Enum
+
+from vaccine.base_application import BaseApplication
+from vaccine.states import (
+    Choice,
+    ChoiceState,
+    EndState,
+    ErrorMessage,
+    FreeText,
+    MenuState,
+)
+from vaccine.utils import luhn_checksum
+
+MAX_AGE = 122
+
+
+class Application(BaseApplication):
+    START_STATE = "state_age_gate"
+
+    class ID_TYPES(Enum):
+        rsa_id = "RSA ID Number"
+        passport = "Passport Number"
+        asylum_seeker = "Asylum Seeker Permit number"
+        refugee = "Refugee Number Permit number"
+
+    async def state_age_gate(self):
+        return MenuState(
+            self,
+            question="\n".join(
+                [
+                    "VACCINE REGISTRATION",
+                    "The SA Department of Health thanks you for helping to defeat the "
+                    "coronavirus!",
+                    "",
+                    "Are you 40 years or older?",
+                ]
+            ),
+            choices=[
+                Choice("state_identification_type", "Yes"),
+                Choice("state_under40_notification", "No"),
+            ],
+            error="Self registration is currently only available to those 40 years or "
```

Comment:
What would trigger this error? invalid input?

Reply:
Yup. So the `MenuState` is very similar to the `ChoiceState`. By default, for `1. Yes`, you can either respond with `1` or `yes`. If you respond with anything else, you'll get the error message, with the list of options.